sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -691,17 +691,191 @@ public virtual void ReadAllEventsComplete(string eventHubName)
         /// <param name="errorMessage">The message for the exception that occurred.</param>
         ///
         [Event(36, Level = EventLevel.Error, Message = "An exception occurred during partition processing or load balancing for processor instance with identifier '{0}' for Event Hub: {1} and Consumer Group: {2}.  Error Message: '{3}'")]
-        public virtual void EventProcessingTaskError(string identifier,
-                                                     string eventHubName,
-                                                     string consumerGroup,
-                                                     string errorMessage)
+        public virtual void EventProcessorTaskError(string identifier,
+                                                    string eventHubName,
+                                                    string consumerGroup,
+                                                    string errorMessage)
         {
             if (IsEnabled())
             {
                 WriteEvent(36, identifier ?? string.Empty, eventHubName ?? string.Empty, consumerGroup ?? string.Empty, errorMessage ?? string.Empty);
             }
         }
 
+        /// <summary>
+        ///   Indicates that an <see cref="EventProcessor{TPartition}" /> instance has taken ownership of a partition and is beginning to process it.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is starting.</param>
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        /// <param name="eventHubName">The name of the Event Hub that the processor is associated with.</param>
+        /// <param name="consumerGroup">The name of the consumer group that the processor is associated with.</param>
+        ///
+        [Event(37, Level = EventLevel.Verbose, Message = "Starting to process partition '{0}' using processor instance with identifier '{1}' for Event Hub: {2} and Consumer Group: {3}.")]
+        public virtual void EventProcessorPartitionProcessingStart(string partitionId,
```

Comment:
What if we ever need to add more parameters to these virtual methods? i.e. would it make sense to use the "args" pattern for them?

Reply:
This wasn't intended to much to capture the arguments for the method but the "identity" of the processor - what is the service context in which it is operating.    The Consumer Group, Event Hub, Partition Grouping is basically the "resource" of the service.    Between that and the identity of the processor, you should know what processor is acting against what service context, so I felt it was important to represent them strongly in the logs as well.